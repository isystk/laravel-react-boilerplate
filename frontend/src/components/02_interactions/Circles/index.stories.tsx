import { Meta, Story } from '@storybook/react'
import React from 'react'
import Circles from './index'

export default {
  title: '02_interactions/Circles',
  component: Circles,
} as Meta

const Template: Story = () => {
  return (
    <Circles>
      <div
        style={{
          height: '600px',
        }}
      ></div>
    </Circles>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
