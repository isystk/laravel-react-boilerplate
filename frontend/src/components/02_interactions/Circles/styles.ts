import { css, keyframes } from '@emotion/css'

export const area = css`
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0;
  margin: 0;
`

export const circles = css`
  position: absolute;
  top: 0;
  left: 0;
  overflow: hidden;
  width: 100%;
  height: 100%;
`

const fadeOutRotation = keyframes`
0% {
  transform: translateY(0) rotate(0deg);
  opacity: 1;
  border-radius: 0;
}
100% {
  transform: translateY(-1000px) rotate(720deg);
  opacity: 0;
  border-radius: 50%;
}
`

const circlesLi = css`
  position: absolute;
  display: block;
  list-style: none;
  width: 20px;
  height: 20px;
  background: #eee;
  animation: ${fadeOutRotation} 25s linear infinite;
  bottom: -150px;
`

export const circlesLi1 = css([
  circlesLi,
  ` left: 25%; width: 80px; height: 80px; animation-delay: 0s; animation-duration: 0s;`,
])
export const circlesLi2 = css([
  circlesLi,
  ` left: 10%; width: 20px; height: 20px; animation-delay: 2s; animation-duration: 12s;`,
])
export const circlesLi3 = css([
  circlesLi,
  ` left: 70%; width: 20px; height: 20px; animation-delay: 4s; animation-duration: 0s;`,
])
export const circlesLi4 = css([
  circlesLi,
  ` left: 40%; width: 60px; height: 60px; animation-delay: 0s; animation-duration: 18s;`,
])
export const circlesLi5 = css([
  circlesLi,
  ` left: 65%; width: 20px; height: 20px; animation-delay: 0s; animation-duration: 0s;`,
])
export const circlesLi6 = css([
  circlesLi,
  ` left: 75%; width: 110px; height: 110px; animation-delay: 3s; animation-duration: 0s;`,
])
export const circlesLi7 = css([
  circlesLi,
  ` left: 35%; width: 150px; height: 150px; animation-delay: 7s; animation-duration: 0s;`,
])
export const circlesLi8 = css([
  circlesLi,
  ` left: 50%; width: 25px; height: 25px; animation-delay: 15s; animation-duration: 45s;`,
])
export const circlesLi9 = css([
  circlesLi,
  ` left: 20%; width: 15px; height: 15px; animation-delay: 2s; animation-duration: 35s;`,
])
export const circlesLi10 = css([
  circlesLi,
  ` left: 85%; width: 150px; height: 150px; animation-delay: 0s; animation-duration: 11s;`,
])
