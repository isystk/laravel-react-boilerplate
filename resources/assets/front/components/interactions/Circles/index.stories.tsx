import Circles from './index';
import type { Meta, StoryFn } from '@storybook/react';

export default {
  title: 'Components/Interactions/Circles',
  component: Circles,
  tags: ['autodocs'],
} as Meta<typeof Circles>;

export const Default: StoryFn = () => (
  <Circles>
    <div style={{ height: '500px' }}>Some content inside Circles component</div>
  </Circles>
);
