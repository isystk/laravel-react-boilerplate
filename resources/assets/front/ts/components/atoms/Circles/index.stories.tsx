import type { Meta, StoryObj } from '@storybook/react';

import Circles from './index';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
    title: 'Components/Atoms/Circles',
    component: Circles,
    parameters: {
        layout: 'centered',
    },
    tags: ['autodocs'],
    argTypes: {
        children: {
            control: 'text',
            description: 'Content inside the Circles component',
        },
    },
    args: {
        children: 'Sample Content',
    },
} satisfies Meta<typeof Circles>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        children: <div style={{ width: '1200px', height: '1000px', backgroundColor: 'black', color: 'white', padding: '20px' }}>Contents</div>,
    },
};
