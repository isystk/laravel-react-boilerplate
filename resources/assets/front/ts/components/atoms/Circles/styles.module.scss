@use "@/styles/variable";

.area {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0;
  margin: 0;
}

.circles {
  position: absolute;
  top: 0;
  left: 0;
  overflow: hidden;
  width: 100%;
  height: 100%;
}

@keyframes fadeOutRotation {
  0% {
    transform: translateY(0) rotate(0deg);
    opacity: 1;
    border-radius: 0;
  }
  100% {
    transform: translateY(-2000px) rotate(720deg);
    opacity: 0;
    border-radius: 50%;
  }
}

.circle {
  position: absolute;
  display: block;
  list-style: none;
  background: variable.$white;
  bottom: -150px;
}

.circlesLi1 {
  @extend .circle;
  left: 25%;
  width: 80px;
  height: 80px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 0s;
  animation-duration: 10s;
}

.circlesLi2 {
  @extend .circle;
  left: 95%;
  width: 20px;
  height: 20px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 2s;
  animation-duration: 12s;
}

.circlesLi3 {
  @extend .circle;
  left: 50%;
  width: 20px;
  height: 20px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 4s;
  animation-duration: 10s;
}

.circlesLi4 {
  @extend .circle;
  left: 40%;
  width: 60px;
  height: 60px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 0s;
  animation-duration: 18s;
}

.circlesLi5 {
  @extend .circle;
  left: 65%;
  width: 20px;
  height: 20px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 0s;
  animation-duration: 20s;
}

.circlesLi6 {
  @extend .circle;
  left: 75%;
  width: 110px;
  height: 110px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 3s;
  animation-duration: 50s;
}

.circlesLi7 {
  @extend .circle;
  left: 35%;
  width: 120px;
  height: 120px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 7s;
  animation-duration: 40s;
}

.circlesLi8 {
  @extend .circle;
  left: 50%;
  width: 25px;
  height: 25px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 15s;
  animation-duration: 25s;
}

.circlesLi9 {
  @extend .circle;
  left: 90%;
  width: 15px;
  height: 15px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 2s;
  animation-duration: 15s;
}

.circlesLi10 {
  @extend .circle;
  left: 5%;
  width: 50px;
  height: 50px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 0s;
  animation-duration: 35s;
}

.circlesLi11 {
  @extend .circle;
  left: 15%;
  width: 40px;
  height: 40px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 0s;
  animation-duration: 30s;
}

.circlesLi12 {
  @extend .circle;
  left: 3%;
  width: 15px;
  height: 15px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 2s;
  animation-duration: 12s;
}

.circlesLi13 {
  @extend .circle;
  left: 92%;
  width: 30px;
  height: 30px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 7s;
  animation-duration: 20s;
}

.circlesLi14 {
  @extend .circle;
  left: 30%;
  width: 30px;
  height: 30px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 0s;
  animation-duration: 18s;
}

.circlesLi15 {
  @extend .circle;
  left: 55%;
  width: 20px;
  height: 20px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 4s;
  animation-duration: 10s;
}

.circlesLi16 {
  @extend .circle;
  left: 85%;
  width: 60px;
  height: 60px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 3s;
  animation-duration: 40s;
}

.circlesLi17 {
  @extend .circle;
  left: 25%;
  width: 60px;
  height: 60px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 7s;
  animation-duration: 20s;
}

.circlesLi18 {
  @extend .circle;
  left: 10%;
  width: 15px;
  height: 15px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 15s;
  animation-duration: 15s;
}

.circlesLi19 {
  @extend .circle;
  left: 10%;
  width: 15px;
  height: 15px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 2s;
  animation-duration: 25s;
}

.circlesLi20 {
  @extend .circle;
  left: 95%;
  width: 50px;
  height: 50px;
  animation: fadeOutRotation 25s linear infinite;
  animation-delay: 0s;
  animation-duration: 35s;
}
