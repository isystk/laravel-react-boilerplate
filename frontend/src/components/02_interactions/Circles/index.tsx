import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import * as styles from './styles'

/** CirclesProps Props */
export type CirclesProps = WithChildren
/** Presenter Props */
export type PresenterProps = CirclesProps

/** Presenter Component */
const CirclesPresenter: FC<PresenterProps> = ({ children, ...props }) => (
  <>
    <div className={styles.area}>
      <ul className={styles.circles}>
        <li className={styles.circlesLi1}></li>
        <li className={styles.circlesLi2}></li>
        <li className={styles.circlesLi3}></li>
        <li className={styles.circlesLi4}></li>
        <li className={styles.circlesLi5}></li>
        <li className={styles.circlesLi6}></li>
        <li className={styles.circlesLi7}></li>
        <li className={styles.circlesLi8}></li>
        <li className={styles.circlesLi9}></li>
        <li className={styles.circlesLi10}></li>
      </ul>
      <div className="relative">{children}</div>
    </div>
  </>
)

/** Container Component */
const CirclesContainer: React.FC<
  ContainerProps<CirclesProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  return presenter({
    children,
    ...props,
  })
}

export default connect<CirclesProps, PresenterProps>(
  'Circles',
  CirclesPresenter,
  CirclesContainer
)
