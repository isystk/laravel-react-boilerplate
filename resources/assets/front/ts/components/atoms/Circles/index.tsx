import styles from './styles.module.scss';
import {ReactNode} from "react";

type Props = {
    children: ReactNode,
};

const Circles = ({ children }: Props) => {
    return (
        <div className={styles.area}>
            <ul className={styles.circles}>
                <li className={styles.circlesLi1}></li>
                <li className={styles.circlesLi2}></li>
                <li className={styles.circlesLi3}></li>
                <li className={styles.circlesLi4}></li>
                <li className={styles.circlesLi5}></li>
                <li className={styles.circlesLi6}></li>
                <li className={styles.circlesLi7}></li>
                <li className={styles.circlesLi8}></li>
                <li className={styles.circlesLi9}></li>
                <li className={styles.circlesLi10}></li>
                <li className={styles.circlesLi11}></li>
                <li className={styles.circlesLi12}></li>
                <li className={styles.circlesLi13}></li>
                <li className={styles.circlesLi14}></li>
                <li className={styles.circlesLi15}></li>
                <li className={styles.circlesLi16}></li>
                <li className={styles.circlesLi17}></li>
                <li className={styles.circlesLi18}></li>
                <li className={styles.circlesLi19}></li>
                <li className={styles.circlesLi20}></li>
            </ul>
            <div className="relative">{children}</div>
        </div>
    );
};

export default Circles;

