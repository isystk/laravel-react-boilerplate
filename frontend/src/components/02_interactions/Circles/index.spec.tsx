import React from 'react'
import renderer from 'react-test-renderer'
import Circles from './index'

describe('Circles', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(
      <Circles>
        <div
          style={{
            height: '600px',
          }}
        ></div>
      </Circles>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
