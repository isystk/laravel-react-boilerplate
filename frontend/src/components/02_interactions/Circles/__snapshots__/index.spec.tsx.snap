// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circles Match Snapshot 1`] = `
<div
  className="css-1wwfi4"
>
  <ul
    className="css-10ffb9g"
  >
    <li
      className="css-i6ammj"
    />
    <li
      className="css-l07tar"
    />
    <li
      className="css-1hre2z7"
    />
    <li
      className="css-p2xyi8"
    />
    <li
      className="css-u8edpl"
    />
    <li
      className="css-ik3prs"
    />
    <li
      className="css-1wh69e5"
    />
    <li
      className="css-1cg3at6"
    />
    <li
      className="css-1iaru34"
    />
    <li
      className="css-1io9drp"
    />
  </ul>
  <div
    className="relative"
  >
    <div
      style={
        {
          "height": "600px",
        }
      }
    />
  </div>
</div>
`;
